packages/client/src/views/SysAdmin/Config/DynamicModal.tsx

```diff
@@ -158,35 +219,66 @@ class DynamicModalComponent extends React.Component<IFullProps, State> {
             this.props.intl.formatMessage(messages.applicationNameChangeError)
           )
         })
+    } else if (
+      modalName === GeneralActionId.GOVT_LOGO &&
+      value.COUNTRY_LOGO?.file &&
+      value.COUNTRY_LOGO?.fileName
+    ) {
+      if (
+        this.isWithinFileLength(value.COUNTRY_LOGO.file as string) === false
+      ) {
+        this.setError(
+          this.props.intl.formatMessage(messages.govtLogoFileLimitError)
+        )
+        this.setState({
+          govtLogo: EMPTY_STRING,
+          logoFile: {
+            name: EMPTY_STRING,
+            type: EMPTY_STRING,
+            data: EMPTY_STRING
+          }
+        })
+        valueChanged(
+          NOTIFICATION_TYPE.ERROR,
+          this.props.intl.formatMessage(messages.govtLogoFileLimitError)
+        )
+      } else {
+        await callUpdateGovtLogoMutation(
+          value.COUNTRY_LOGO.file,
+          value.COUNTRY_LOGO.fileName,
+          this.props,
+          this.setUpdatingValue,
+          this.setError
+        )
+          .then(() => {
+            valueChanged(
+              NOTIFICATION_TYPE.SUCCESS,
+              this.props.intl.formatMessage(messages.govtLogoChangeNotification)
+            )
+          })
+          .catch(() => {
+            this.setState({
+              errorOccured: true,
+              errorMessages: this.props.intl.formatMessage(
+                messages.govtLogoChangeError
+              )
+            })
+            valueChanged(
+              NOTIFICATION_TYPE.ERROR,
+              this.props.intl.formatMessage(messages.govtLogoChangeError)
+            )
+          })
```

Comment:
Instead of then catch block use await

```suggestion
       try{
    await callUpdateGovtLogoMutation(
        value.COUNTRY_LOGO.file,
        value.COUNTRY_LOGO.fileName,
        this.props,
        this.setUpdatingValue,
        this.setError)
    valueChanged(
            NOTIFICATION_TYPE.SUCCESS,
            this.props.intl.formatMessage(messages.govtLogoChangeNotification)
        )
}catch{
    this.setError(
        this.props.intl.formatMessage(messages.govtLogoChangeError)
    )
    valueChanged(
        NOTIFICATION_TYPE.ERROR,
        this.props.intl.formatMessage(messages.govtLogoChangeError))
}
```